### ISSUE TITLE
Use alternative SPARQL parser

### ISSUE THREAD

**Issue by rubensworks** [comunica/comunica#403]
#### Issue type:

- :heavy_plus_sign: Feature request

____
#### Description:

SPARQL.js is currently being used for parsing SPARQL queries. This library works great, but is unfortunately quite big when webpacked. For this reason (and also lack of modularity), we may want to look into implementing a custom parser.

Edit: included from @jeswr's issue #1018:

There are some use-cases in which it would be beneficial to process SPARQL queries in a stream-like manner. For instance an INSERT DATA query containing a large number of triples to insert would be better processed in real-time rather than needing to buffer all the changes in memory before making the update.

One way of achieving this would be to create a new algebra library where property access on the algebra return promises that resolve when processing has completed, and anything that was an array in sparqlalgebrajs becomes an AsyncIterable. Consequently it could potentially be worth looking into this when work on supporting SPARQL 1.2 starts.

**Comment by joachimvh** [comment]
This is actually a sparqlalgebra.js issue since that's the only thing that makes use of it I think?

**Comment by rubensworks** [comment]
Indeed, kind of. But it can be developed independently though. Also, if this new parser would output parsed queries in the same format, then sparqlalgebra.js could simply plug in this new parser.

**Comment by rubensworks** [comment]
Related to #386

**Comment by rubensworks** [comment]
@jitsedesmet is working on this in https://github.com/comunica/traqula, and is already being used in https://github.com/comunica/comunica/tree/feature/sparql12 (which will eventually get merged into master)
